doc/governance/governance.rst

```diff
@@ -122,3 +122,62 @@ The following repositories are managed by the ros-controls PMC:
+The stable release, called **released ros2_control version** from now, will be announced on `ROS discourse <https://discourse.openrobotics.org/c/ros-controls/ros-controls-announce-news/107>`__ on time.
+
+Versioning
+~~~~~~~~~~
+
+We will use the ROS-specific rules on top of ``semver's`` for versioning, but also adhere to some ros-controls-specific rules:
+
+* Major version increments (i.e. breaking changes) should not be made within a **released ros2_control version**.
+
+* ros2_control heavily relies on the usage of `pluginlib <https://index.ros.org/p/pluginlib/>`__. Therefore, we distinguish two types of compiled code: non-plugin code together with plugin base classes, and plugins itself.
+
+  * ABI of plugins may change at every release, i.e., also within **released ros2_control version**. Typically, they are built by the buildfarm and not intended to be linked against your code.
+  * For non-plugin code, the ABI is considered part of the public interface. Any change that requires recompiling dependent code is considered major (breaking) and must not be made within a **released ros2_control version**.
+
+Public API declaration
+~~~~~~~~~~~~~~~~~~~~~~
+
+According to ``semver``, every package must clearly declare a public API.
+We will use the "Public API Declaration" section of the quality declaration of a package to declare what symbols are part of the public API.
+
+For most C and C++ packages the declaration is any header that it installs.
+However, it is acceptable to define a set of symbols which are considered private.
+Avoiding private symbols in headers can help with ABI stability, but is not required.
+
+For other languages like Python, a public API must be explicitly defined, so that it is clear what symbols can be relied on with respect to the versioning guidelines.
+The public API can also be extended to build artifacts like configuration variables, CMake config files, etc. as well as executables and command-line options and output.
+Any elements of the public API should be clearly stated in the package's documentation.
+If something you are using is not explicitly listed as part of the public API in the package's documentation, then you cannot depend on it not changing between minor or patch versions.
+
+Deprecation strategy
+~~~~~~~~~~~~~~~~~~~~
+
+Where possible, we will use the tick-tock deprecation and migration strategy for breaking changes (API or behavior-breaking changes).
+
+* New deprecations can be run-time messages or compiler warnings expressing that the functionality is being deprecated. The functionality will be completely removed in any future release, or at latest in the next **released ros2_control version** (there may be details in the deprecation note).
+
+* New deprecations can also come in every release of **released ros2_control version** by performing backports of any changes to the rolling version. But the deprecated functionality will remain available.
```

Comment:
```suggestion
* New deprecations can also come in every release of **released ros2_control version** by performing backports of any changes to the rolling version. These are meant to help users migrate early, however the functionality will remain available in that release.
```

Reply:
what do you mean with "release"? A release is a tag, pushed with bloom-release to the buildfarm.

Reply:
May be he meant within the same ROS Distro version